community/cypher/cypher-compiler-3.0/src/test/scala/org/neo4j/cypher/internal/compiler/v3_0/planner/logical/plans/IndexContainsScanLeafPlannerTest.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.compiler.v3_0.planner.logical.plans
+
+import org.neo4j.cypher.internal.compiler.v3_0.planner.BeLikeMatcher._
+import org.neo4j.cypher.internal.compiler.v3_0.planner._
+import org.neo4j.cypher.internal.compiler.v3_0.planner.logical.steps.indexContainsScanLeafPlanner
+import org.neo4j.cypher.internal.frontend.v3_0.ast._
+import org.neo4j.cypher.internal.frontend.v3_0.test_helpers.CypherFunSuite
+
+class IndexContainsScanLeafPlannerTest extends CypherFunSuite with LogicalPlanningTestSupport2 {
```

Comment:
Add test for how to behave in the presence of a hint, but index missing.
Add test for how to behave when there are two predicates and several possible indexes to pick.

Also, related to comment above, what about a unique and normal index on the same label/property pair?